File: mixtourTests/core/MIXCoreBoardTests.m
//
//  MIXCoreBoardTest.m
//  mixtour
//
//  Created by Carsten Wenderdel on 2013-06-17.
//  Copyright (c) 2013 Carsten Wenderdel. All rights reserved.
//

#import <XCTest/XCTest.h>

#import "MIXCoreBoard.h"

@interface MIXCoreBoardTest : XCTestCase

@end

@implementation MIXCoreBoardTest


- (void)testSize {
    struct MIXCoreBoard testStruct;
    XCTAssertEqual((size_t)60, sizeof(testStruct), @"core board has not proper size");
}


@end

File: mixtourTests/core/MIXCoreBoardTests.swift
//
//  MIXCoreBoardTests.swift
//  mixtour
//
//  Created by Wenderdel, Carsten on 30/12/14.
//  Copyright (c) 2014 Carsten Wenderdel. All rights reserved.
//

import XCTest
@testable import mixtour

class MIXCoreBoardTests : XCTestCase {
    
    func testBoard() {
        // Not sure about the difference between strideof and sizeof. Currently Swift documention is not clear enough about it. So better use both in unit test - 64 bytes for C struct!

        XCTAssertEqual(64, MemoryLayout<MIXCoreBoard>.size)
        XCTAssertEqual(64, MemoryLayout<MIXCoreBoard>.stride)
        XCTAssertEqual(4, MemoryLayout<MIXCoreBoard>.alignment)
    }
}

